### DISCUSSION TITLE
Instructing fpm not to include a folder

### DISCUSSION THREAD

**Discussion by ivan-pi** [fortran-lang/fpm#474] [General]
Is their currently a method to prevent `fpm` from searching a folder in the source directory? I would like to archive some legacy routines in the `src/legacy/` folder, but don't want them to be part of the output library.
```
.
├── fpm.toml
├── LICENSE
├── README.md
└── src
    ├── legacy
    │   ├── back.f
    │   ├── lu.f
    │   └── stiff3.f
    ├── linalg.f90
    ├── robertson.f90
    ├── stiff3_solver.f90
    └── vanpol.f90
```

Would it be better practice to move the `legacy` folder out of `src`?

**Comment by awvwgk** [comment]
I found myself using the below directory layout for fpm quite a lot (mainly due to the initial Haskell fpm constraint of the module namespaces)

```
.
├── fpm.toml
└── src
    ├── <name>
    │   └── ...
    └── <name>.f90
```

The toplevel `<name>.f90` is usually just the public API reexport and the actual source is in `<name>` ([example](https://github.com/dftd4/dftd4/tree/main/src)).
Without the toplevel module (maybe moving it to `src/<name>/<name>.f90`) one could change the library subdirectory to

```toml
[library]
source-dir = "src/<name>"
```

and fpm would ignore other subdirectories in `src/`.
